modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts

```diff
@@ -52,19 +52,24 @@ export const detect = (popupSinkElem: SugarElement<HTMLElement>): Optional<Scrol
 export const detectWhenSplitUiMode = (editor: Editor, popupSinkElem: SugarElement<HTMLElement>): Optional<ScrollingContext> =>
   Options.isSplitUiMode(editor) ? detect(popupSinkElem) : Optional.none();
 
+const isFullscreen = (editor: Editor): boolean =>
+  editor.plugins.fullscreen && editor.plugins.fullscreen.isFullscreen();
+
 // Using all the scrolling viewports in the ancestry, limit the absolute
 // coordinates of window so that the bounds are limited by all the scrolling
 // viewports.
-export const getBoundsFrom = (sc: ScrollingContext): Bounds => {
+export const getBoundsFrom = (editor: Editor, sc: ScrollingContext): Bounds => {
```

Comment:
It looks like the only reason you have `editor` here is for the `fullscreen` call? That's too tightly coupled.

Please take some inspiration from other parts of our UI architecture and pass an `isFullscreen` function to this method instead of the full editor object. Reducing the capability of function arguments is an easy way to avoid spaghetti code dependencies.

Or, if you look really closely, there's a way to completely avoid adding the extra method argument. I'm happy to chat if you can't figure it out.

- [ ] Improve method architecture

Reply:
ok, I think I got it, I moved the check inside `detectWhenSplitUiMode`, but after that I had to refactor the code in `Theme.ts` (and also the comment of that code :sweat_smile: ), let me know this is what you had in mind or I'm off track

Reply:
That's not what I expected, but it's probably cleaner.

What I was thinking is the `sc` parameter here is created in a context that has an editor, so you could add an `isFullscreen` property to it:
https://github.com/tinymce/tinymce/blob/64333acf0e9e93aa1853c824405654fb663e2e28/modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts#L9-L12

I'm not entirely convinced that checking for a scrollable parent every time render is called is a good performance decision, but so long as it only happens when a popup opens it should be fine.